ledger-api/grpc-definitions/com/daml/ledger/api/v1/error.proto

```diff
@@ -0,0 +1,130 @@
+//    Other forms of error details may be added in the future though.
+//
+//
+// Common error codes and details for all services
+// ***********************************************
+//
+// Individual services may use the error codes below for other cases too.
+// See the documentation of the service for details.
+//
+// ``UNAUTHENTICATED``:
+// - The request does not include a valid access token.
+//
+// ``PERMISSION_DENIED``:
+// - The claims in the token are insufficient to perform a given operation.
+//
+// ``NOT_FOUND``:
+// - The request does not include a valid ledger id.
+//
+// ``INVALID_ARGUMENT``:
+// - The payload is malformed or is missing required fields.
+//
+// ``RESOURCE_EXHAUSTED``:
+// - The participant or the ledger is overloaded. Clients should back off exponentially and retry.
+//
+// ``DEADLINE_EXCEEDED``:
+// - The server failed to produce a result within the given timeouts.
+//   For requests that change the ledger state, clients MUST assume neither
+//   that the request succeeded nor that it was rejected.
+//
+//
+// Error codes for command submissions
+// ***********************************
+//
+// TODO(v2) Define test cases for these errors and enforce these error rules via the ledger API test tool
+//
+// This applies to the services in ``command_service.proto`` and ``command_submission_service.proto``.
+// The common error codes from above also apply here.
+// The status message in command completions as defined in ``completion.proto`` follow the same rules.
+//
+// Some Daml ledger implementations may produce the same error codes in further situations.
+// See the documentation of the specific Daml ledger implementation.
+//
+// If the submission led to a completion event, the following error details are provided:
+// 1. If the error is reported outside of the completion event (e.g., via the command service),
+//    the metadata key ``completion_offset`` contains the corresponding completion offset.
+//    This offset can be used as a submission rank in later calls.
+// 2. If the completion event at the offset is a definite answer for the deduplication
+//    and rank guarantees described in ``commands.proto``,
+//    the metadata key ``definite_answer`` is set to ``true``.
+//
+// ``UNAVAILABLE``:
+// - The participant is not yet ready to submit commands or if the service has been shut down.
+//
+// ``RESOURCE_EXHAUSTED``:
+// - The number of in-flight commands reached the maximum (if a limit is configured).
+//
+// ``NOT_FOUND``:
+// - The submission rank or deduplication period are too old.
+//
+// ``OUT_OF_RANGE``
+// - The deduplication period starts in the future (e.g., the specified offset is higher than the completion end).
+// - The submission rank is after the completion end.
+//
+// ``INVALID_ARGUMENT``:
+// - A party involved in the transaction is not known on the ledger.
+// - The interpreted command is inconsistent with the state of the ledger.
```

Comment:
I think this needs to be defined more precisely. Probably this PR is not the right place for doing that though.

Reply:
That's why there's the `TODO(error cleanup)` in the next line. As the PR description says, this needs to be worked out in detail, but not here.